src/wayland/shm/mod.rs

```diff
@@ -219,6 +219,35 @@ where
     }
 }
 
+/// Call given closure with the contents of the given buffer for mutable access
+///
+/// If the buffer is managed by the provided `ShmGlobal`, its contents are
+/// extracted and the closure is extracted with them:
+///
+/// - The first argument is a data slice of the contents of the pool
+/// - The second argument is the specification of this buffer is this pool
+///
+/// If the buffer is not managed by the provided `ShmGlobal`, the closure is not called
+/// and this method will return `Err(BufferAccessError::NotManaged)` (this will be the case for an
+/// EGL buffer for example).
+pub fn with_buffer_contents_mut<F, T>(buffer: &wl_buffer::WlBuffer, f: F) -> Result<T, BufferAccessError>
```

Comment:
So we want to guard against the future access flags in wl_shm V3? Not every error will end up being NotManaged in the future

Reply:
Do you have some reference, what might happen in the future?

Reply:
https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/161/ is the proposal

eli5: In version 3, a new request is added which allows specifying what the server can do with the buffer (read and or write). For v2 and v1, assume the buffer is always readable and writable.

I believe there are some semantics changing related to resizing, but those are out to scope for this pull request.

Reply:
Ok, so we want a `not-writable` error type, got it.

Reply:
Added new Error-variants for non-readable or non-writable buffers.